wasm/README.md

```diff
@@ -1,14 +1,225 @@
-## WASM based javascript wrapper for DIDComm
+# DIDComm JS
 
-### 🛠️ Build with `wasm-pack build`
+[![License](https://img.shields.io/badge/License-Apache%202.0-blue.svg)](https://opensource.org/licenses/Apache-2.0)
+[![Unit Tests](https://github.com/sicpa-dlab/didcomm-rust/workflows/verify/badge.svg)](https://github.com/sicpa-dlab/didcomm-rust/actions/workflows/verify.yml)
+
+Basic [DIDComm v2](https://identity.foundation/didcomm-messaging/spec) support for modern browsers and NodeJS.
+
+## Under the hood
+
+This package is written in Rust using [didcomm](<(https://img.shields.io/crates/v/didcomm)>) crate. It compiles
+to `wasm32` and exposes Javascript/Typescript API with [wasm-bindgen](https://github.com/rustwasm/wasm-bindgen) help.
+Also [wasmp-pack](https://github.com/rustwasm/wasm-pack) helps in packaging and publishing.
+
+## Usage
+
+To use `didcomm-js` install it with npm
+
+```sh
+npm install didcomm --save # If you plan use webpack or other bundler
+
+npm install didcomm-node --save # If you plan use it without bandlers in NodeJS
+
+```
+
+**TODO: Exact package name will be defined after first release**
+
+## Run demo
+
+```sh
+wasm-pack build --target nodejs # builds NodeJS package in pkg directiry
+cd ./demo
+npm install
+npm run start
+```
+
+## Assumptions and Limitations
+
+- This library requires `wasm32` compatible environment (modern browsers and recent NodeJS are supported).
+- In order to use the library, `SecretsResolver` and `DIDResolver` interfaces must be implemented on the application level.
+  Demo application provides 2 simple implementations `ExampleDIDResolver`
+  and `ExampleSecretsResolver` that allows resolve locally known DID docs and secrets for tests/demo purposes.
+- Verification materials are expected in JWK.
+- Key IDs (kids) used in `SecretsResolver` must match the corresponding key IDs from DID Doc verification methods.
+- Key IDs (kids) in DID Doc verification methods and secrets must be a full [DID Fragment](https://www.w3.org/TR/did-core/#fragment), that is `did#key-id`.
+- Verification methods referencing another DID Document are not supported (see [Referring to Verification Methods](https://www.w3.org/TR/did-core/#referring-to-verification-methods)).
+- The following curves and algorithms are supported:
+  - Encryption:
+    - Curves: X25519, P-256
+    - Content encryption algorithms:
+      - XC20P (to be used with ECDH-ES only, default for anoncrypt),
+      - A256GCM (to be used with ECDH-ES only),
+      - A256CBC-HS512 (default for authcrypt)
+    - Key wrapping algorithms: ECDH-ES+A256KW, ECDH-1PU+A256KW
+  - Signing:
+    - Curves: Ed25519, Secp256k1, P-256
+    - Algorithms: EdDSA (with crv=Ed25519), ES256, ES256K
+- DIDComm has been implemented under the following [Assumptions](https://hackmd.io/i3gLqgHQR2ihVFV5euyhqg)
+
+### **Features that will be supported in next versions**
```

Comment:
We've already added support for base58 and multibase on the core level. Does it need to be ported to didcomm JS, or it's already there?